### ISSUE TITLE
ESP32 S3 cannot use this library file,could you tell me where the problem is

### ISSUE THREAD

**Issue by andylu321123** [techtoys/Arduino-Memory-LCD-Driver#3]
sir,

 Thank you very much for your contribution.

I have two ESP32 development boards，esp-wroom-32 and esp32 s3. esp-wroom-32  can be used very well.
but  esp32 S3 , I can't use it. I confirm I need modify SPI pin definitions for adapt to ESP32S3. but The result was disappointing.
 
I am trying to redefine the pins,But I failed. 

Could you guide me on how to modify the code to adapt to it?

//#define VSPI_MOSI  40
//#define VSPI_SCK  39
//#define SCS   38

or  
//#define SS      42 
//#define MOSI    35 
//#define SCK     36 

bye

**Comment by techtoys** [comment]
ESP32S3 uses GPIO matrix for its peripherals including SPI. That means you may assign any pin to SPI.
You may try this:
1) Locate MemoryLCD.cpp file
2) Search `hal_bsp_init(void)`
3) There is a line `_SPI->begin()`. Change it to `_SPI->begin(your_sclk_pin, -1, your_mosi_pin, -1)`.
The second -1 indicates miso which is not required for memory LCD. The last -1 is the CS pin which we are using manual GPIO toggle. That pin is not needed either.
Don't forget `your_sclk_pin` & `your_mosi_pin` must be output pins.

Recompile and see...

Hope it helps.

**Comment by andylu321123** [comment]
thanks for your help, I had success

**Comment by vv96ll** [comment]
Hello, can you share how you used the esp32s3 to light up this screen?

**Comment by techtoys** [comment]
Honestly, I have never tried ESP32S3. I did it with ESP32 and Arduino M0 only. I believe ESP32S3 is similar. There was an issue and solved reported by [andylu321123 ](https://github.com/techtoys/Arduino-Memory-LCD-Driver/issues/3#issuecomment-1953248565). Maybe Andy will help.

**Comment by vv96ll** [comment]
Thank you for your reply. According to the above operation, the display is not normal. I can only hope that andylu321123 can reply

**Comment by andylu321123** [comment]
[FirstPixel.zip](https://github.com/techtoys/Arduino-Memory-LCD-Driver/files/14458452/FirstPixel.zip)
![image](https://github.com/techtoys/Arduino-Memory-LCD-Driver/assets/160379392/0e2b5004-5c79-444e-9f01-6d537fdca05a)
![image](https://github.com/techtoys/Arduino-Memory-LCD-Driver/assets/160379392/f2ba9f88-c8aa-4dbb-9715-e32241dc4604)
I hope this documet help you.

**Comment by andylu321123** [comment]
![image](https://github.com/techtoys/Arduino-Memory-LCD-Driver/assets/160379392/cdfbcb3b-f313-4206-bedc-82842b0c3b22)

**Comment by andylu321123** [comment]
Just connet sclk  si scs.  disp and extcomin not need connect .

**Comment by vv96ll** [comment]
Thank you very much for your reply. At first, I thought it was the problem of the pin. I have tried many times, but now the picture is still not normal, Thanks again for your reply.
![微信图片_20240301165108](https://github.com/techtoys/Arduino-Memory-LCD-Driver/assets/81518771/77af5a0c-e964-4753-af31-2d7fe03e7c0b)

**Comment by andylu321123** [comment]
I dont know the reason. Sorry ,but I think the style is not same.
![image](https://github.com/techtoys/Arduino-Memory-LCD-Driver/assets/160379392/f44cf3da-3d90-445f-b70b-a02d05ff9475)

**Comment by andylu321123** [comment]
you can try to run this word.
GFXDisplayPutPixel(10, 10, BLACK); 
then to find how to display,what happed on the screen. then ..... 
maybe you need modify something for display.
I just help you do this. I am not Professionally.

**Comment by vv96ll** [comment]
Yes, there is A difference, my hand is LS027B7DH01A, the normal is not with the suffix A,thanks for your reply.

<img width="430" alt="image" src="https://github.com/techtoys/Arduino-Memory-LCD-Driver/assets/81518771/8c9cc5dc-aa70-46d5-b4fc-ea29e1b1bbce">

**Comment by andylu321123** [comment]
If you may drive LS027B7DH01A nomally,Please let me know,I want to learn it also.

**Comment by vv96ll** [comment]
GFXDisplayPutPixel(10, 10, BLACK);
this word is valid.  be shown in about(11, 10).

I wish I could drive LS027B7DH01A nomally.

![2701d796667ef3a158341f3833dd7c3](https://github.com/techtoys/Arduino-Memory-LCD-Driver/assets/81518771/c6057b05-8450-4b01-b7e3-e3c22820de00)

**Comment by andylu321123** [comment]
Try different points, Find patterns, then modify "void GFXDisplayPutPixel(uint16_t x, uint16_t y, COLOR color)" and so on.  Good luck!!!
